### PR TITLE
feat: add tailnet ValidateVersion

### PR DESCRIPTION
**Pull request by spikecurtis**
Part of #10532

Adds a method to validate a requested version of the tailnet API

### CODE REVIEW

**Comment by spikecurtis** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #11222** <a href="https://app.graphite.dev/github/pr/coder/coder/11222?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #11223** <a href="https://app.graphite.dev/github/pr/coder/coder/11223?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #11224** <a href="https://app.graphite.dev/github/pr/coder/coder/11224?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/coder/coder/11223?utm_source=stack-comment).

File: tailnet/service_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		{
+			name:    "TooNewMinor",
+			version: fmt.Sprintf("%d.%d", tailnet.CurrentMajor, tailnet.CurrentMinor+1),
+		},
+		{
+			name:    "TooNewMajor",
+			version: fmt.Sprintf("%d.%d", tailnet.CurrentMajor+1, tailnet.CurrentMinor),
+		},
+		{
+			name:      "1.0",
```
Comment (deansheather) [inline tailnet/service_test.go:33]: 2.0 too

File: tailnet/service_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+			version: "cats.dogs",
+		},
+		{
+			name:    "Malformed2",
+			version: "1.0.1",
+		},
+		{
+			name:    "Malformed3",
+			version: "11",
+		},
```
Comment (deansheather) [inline tailnet/service_test.go:52]: Should be some TooOld tests, I guess those would be `0.0` for now?

**Review by deansheather — APPROVED** [review]

**Comment by spikecurtis** [comment]
### Merge activity

* **Dec 15, 2:38 AM**: [Graphite](https://app.graphite.dev/github/pr/coder/coder/11223) rebased this pull request as part of a merge.

**Pull request merged**